src/violation.rs

```diff
@@ -50,6 +50,7 @@ impl fmt::Display for Violation {
 #[derive(Debug, Clone)]
 pub enum ViolationCode {
     NoTypesMatched,
+    InvalidNull,
```

Comment:
minor suggestion—it would be good to add a comment indicating how/when this is to be used. Some of the other `ViolationCode`s are pretty self-explanatory, but I think this one is not.